src/components/dialog/use-dialog-appearance.ts

```diff
@@ -0,0 +1,31 @@
+import { RefObject, useEffect, useRef } from 'react'
+
+interface UseDialogAppearanceOptions {
+  isOpen: boolean
+  onClose: () => void
+}
+
+export const handleDialogAppearance =
+  (options: UseDialogAppearanceOptions, ref: RefObject<HTMLDialogElement>) => () => {
+    const { isOpen, onClose } = options ?? {}
+    if (!ref.current || !onClose) return
+
+    if (isOpen) {
+      ref.current.showModal()
+      ref.current.addEventListener('close', onClose)
+    } else {
+      ref.current.removeEventListener('close', onClose)
+      ref.current.close()
+    }
+
+    return () => {
+      ref.current?.removeEventListener('close', onClose)
+    }
+  }
+export const useDialogAppearance = ({ isOpen, onClose }: UseDialogAppearanceOptions) => {
+  const ref = useRef<HTMLDialogElement>(null)
+
+  useEffect(handleDialogAppearance({ isOpen, onClose }, ref), [isOpen, onClose])
```

Comment:
**issue (non-blocking)**: At first glance, I thought this was calling `handleDialogAppearance` immediately, only to realise `handleDialogAppearance` returns a function. I find this a bit confusing, though I really like how you have a descriptive function name with `useEffect` rather than what you normally see (`useEffect(() => { ... }, [...])`).

**suggestion:** What the Console Cloud FE team typically do these days is simply use a named function:

```javascript
useEffect(function handleDialogAppearance() {
  ...
}, [...])
```

This maintains the benefit of a well-named function for the effect, without the awkwardness of needing `handleDialogAppearance` to return a function like it currently does.

Reply:
that's acceptable